### PR TITLE
Changed nullable boolean parameter isFullPath to enum.

### PR DESCRIPTION
**Pull request by alphaleonis**
In the process refactored out a bit of duplicated code that managed this parameter, and changed the signature of some (internal) methods from accepting a large number of bool parameters to accepting a single enum value which improves readability. 

Issue #13 

Care to review this @Yomodo ?

### CODE REVIEW

**Comment by Yomodo** [comment]
I would still prefer PathFormat.Unicode instead of PathFormat.ExtendedLength.

**Comment by alphaleonis** [comment]
Yeah, I don't like the names I chose here really either. But Unicode seems so strange, because it really has nothing to do with the encoding of it, all strings are Unicode in .NET. ;) 
Looking again at MSDN docucumentation of the [GetLongPathName](http://msdn.microsoft.com/en-us/library/windows/desktop/aa364980%28v=vs.85%29.aspx) function, I'd say that "Long" is probably a better word though. 

Also `Standard` should probably include the word *Full* since it should not be a relative path if I understand things correctly it must not be a relative path in this case. 

So perhaps something like `Auto`, `FullShort`, `FullLong`

Or `Auto`, `ShortFullPath`, `LongFullPath`?

File: AlphaFS/Filesystem/DirectoryInfo.cs [hunk 1]
```diff
@@ -1011,8 +989,8 @@ public CopyMoveResult CopyTo1(string destinationPath, CopyOptions copyOptions, C
       public DirectoryInfo CopyTo1(string destinationPath)
       {
          string destinationPathLp;
-         CopyToMoveToInternal(destinationPath, CopyOptions.FailIfExists, null, null, null, out destinationPathLp, false);
-         return new DirectoryInfo(Transaction, destinationPathLp, null);
+         CopyToMoveToInternal(destinationPath, CopyOptions.FailIfExists, null, null, null, out destinationPathLp, PathFormat.Auto);
```
Comment (Yomodo) [inline AlphaFS/Filesystem/DirectoryInfo.cs]: false, auto

File: AlphaFS/Filesystem/DirectoryInfo.cs [hunk 2]
```diff
@@ -771,7 +765,7 @@ public FileSystemInfo[] GetFileSystemInfos(string searchPattern, SearchOption se
       public void MoveTo(string destinationFullPath)
       {
          string destinationPathLp;
-         CopyToMoveToInternal(destinationFullPath, null, MoveOptions.None, null, null, out destinationPathLp, true);
+         CopyToMoveToInternal(destinationFullPath, null, MoveOptions.None, null, null, out destinationPathLp, PathFormat.Auto);
```
Comment (Yomodo) [inline AlphaFS/Filesystem/DirectoryInfo.cs]: true, auto

**Comment by Yomodo** [comment]
How about: `RelativePath`, `FullPath`, `LongFullPath`

**Comment by alphaleonis** [comment]
Or perhaps take inspiration from [UriKind](http://msdn.microsoft.com/en-us/library/system.urikind%28v=vs.110%29.aspx) and use something like `AbsoluteOrRelative`, `Absolute`, `LongAbsolute`?  (Naming things is probably the hardest part of programming btw ;)

**Comment by alphaleonis** [comment]
On second though maybe `Absolute` is not a good term, since a path can probably be *aboslute* without being a *full* path?  A path such as `"\Windows\notepad.exe"` is probably *absolute* but it lacks the drive so... your suggestion is probably the best, though I would probably want to change it to: `RelativeOrFullPath`, `FullPath`, `LongFullPath` since the path may be a full path in the first case as well.

**Comment by Yomodo** [comment]
Rather "RelativePath" than "RelativeOrFullPath", since the latter is kind of confusing?
E.g.: When/what is it: RelativeOrFullPath or FullPath

I think that if we call it RelativePath at least it is clear that AlphaFS does NOT consider it a full path (C:\Windows VS \Windows) by default and thus GetFullPath() will be called.

**Comment by alphaleonis** [comment]
Yeah, but with "RelativePath" it feels as if you cannot specify a full path. Eg. if you don't know whether the path is a full or a relative path, which one should you use?  That's why I went with the UriKind-style of saying that this can be either a relative or a full path, much like UriKind.AbsoluteOrRelative that exists in the .NET framework.

**Comment by Yomodo** [comment]
When implementing the isFullPath parameter, my philosophy was that by default AlphaFS will regard all given paths as relative: always call GetFullPath(),

UNLESS one would specifically specify that the path is a full path or long full path:
isFullPath = true, null respectively.

**Comment by alphaleonis** [comment]
Yeah, I see your point. I guess the thing is that you will probably rarely specify that option anyway, since in those cases you will use an overload that doesn't have that parameter.

But I'll change it back to "Relative" then.

**Comment by Yomodo** [comment]
Well, one, or two places where you definitely want to use a "manual override" is when
dealing with files and folders that end in a trailing space or dot.
(p.s. I'm reviewing as we speak)

**Comment by alphaleonis** [comment]
Ah, okay, yeah, that makes sense.

**Comment by Yomodo** [comment]
Which reminds me: Some CodePlex changesets ago I removed "isFullPath" for the AlphaFS implementations of the .NET methods: Directory.GetXxx() and Directory.EnumXxx() since the number
of overloaded methods was getting out of hand.

Directory.EnumerateFileSystemEntryInfos() is a worthy replacement which can handle the above GetXxx() and EnumXxx() methods.

**Comment by alphaleonis** [comment]
Okay.

**Comment by Yomodo** [comment]
AlphaFS implementation of DirectoryInfo.FullName ends up with a trailing backslash when it should not. Will track this down first.

**Comment by alphaleonis** [comment]
Alright, I'm holding off any further work for now until we can get this and the next one merged in, since there will be too much hassle merging things otherwise.  Let me know if you need me to look into any issues.

**Pull request merged**